### PR TITLE
[NOTIFICATIONS #11] feat(events): fire on-chain transactions events

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Following up https://github.com/cowprotocol/cowswap/pull/4039

1. Added new event `ON_ONCHAIN_TRANSACTION` which contains all necessary information about a on-chain transaction (`OnTransactionPayload`)
2. Added `OnchainTransactionEventsUpdater` which listens `ON_ONCHAIN_TRANSACTION` event and:
 - displays a snackbar in CowSwap UI
 - fires `ON_TOAST_MESSAGE` event
 - plays mooo sound when transaction failed

image

  # To Test

1. All cases from https://github.com/cowprotocol/cowswap/pull/4038
2. The same cases, but through https://github.com/cowprotocol/cowswap/pull/4023 in Dapp mode. Notifications should be displayed on the widget configurator side

### CODE REVIEW

**Review by elena-zh — COMMENTED** [review]
Hey @shoom3301 , here I can see all the same issues reported in https://github.com/cowprotocol/cowswap/pull/4038#pullrequestreview-1937137287

Plus for snackbars it would be great to add 'Fail' word into the motificaiton when wrap/unwrap/approve or other onchain TX fails
fail

**Comment by shoom3301** [comment]
>Plus for snackbars it would be great to add 'Fail' word into the motificaiton when wrap/unwrap/approve or other onchain TX fails

@elena-zh unfortunately I can't do it right now because we need to merge everything first.
Anyway, this is good to have feature, I would keep for the future.

**Comment by elena-zh** [comment]
> > Plus for snackbars it would be great to add 'Fail' word into the motificaiton when wrap/unwrap/approve or other onchain TX fails
> 
> @elena-zh unfortunately I can't do it right now because we need to merge everything first. Anyway, this is good to have feature, I would keep for the future.

Opened #4049 issue for this

**Review by elena-zh — APPROVED** [review]
Approving (with issues that will be fixed in #4038) and #4049 enhancement mentioned above.

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/onchainTransactions/updaters/FinalizeTxUpdater.tsx [hunk 1]
```diff
@@ -373,5 +374,9 @@ export function FinalizeTxUpdater(): null {
     getTwapOrderById,
   ])
 
-  return null
+  return (
+    <>
+      <OnchainTransactionEventsUpdater />
+    </>
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/onchainTransactions/updaters/FinalizeTxUpdater.tsx]: Not necessary, removed

**Pull request merged**